### ISSUE TITLE
Base64 strings as sprites

### ISSUE THREAD

**Issue by ran4erep** [4ian/GDevelop#3984]
## Description

It would be nice if we could add sprites not only from file system or from URL, but from Base64 strings as well

**Comment by arthuro555** [comment]
Did you try using a [data url](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs)? As long as URLs are supported you should be able to use Base64 strings as well using those.

**Comment by ran4erep** [comment]
oh, it's already exist, nice. Then you should add info about this possibility into description block when adding a file from URL

**Comment by AlexandreSi** [comment]
Thanks @arthuro555 for the suggestion!
I'll see how we can add info about this possibility!

**Comment by arthuro555** [comment]
Personally, I believe that wanting to use base64 encoded images is something a very select few users would ever want to use, and if a user even knows about base64 encoded images I would expect them to know what data URLs are too. I feel like including this in the IDE would be giving users an unnecessary detail that would confuse most of them. One of GDevelop's goal being to be easy to understand and accessible, I don't think we should add such technical details in the description of the resource URL selection field like suggested.

**Comment by ran4erep** [comment]
maybe in desktop version only few users will be using Base64 images, but in browser version it's more preferable over the usual URL's. Because no need to find a hosting for images to keep them for your project

**Comment by AlexandreSi** [comment]
> Personally, I believe that wanting to use base64 encoded images is something a very select few users would ever want to use, and if a user even knows about base64 encoded images I would expect them to know what data URLs are too. I feel like including this in the IDE would be giving users an unnecessary detail that would confuse most of them. One of GDevelop's goal being to be easy to understand and accessible, I don't think we should add such technical details in the description of the resource URL selection field like suggested.

I agree with you. It could be a button `Advanced options` at the lower left corner of the dialog.

> maybe in desktop version only few users will be using Base64 images, but in browser version it's more preferable over the usual URL's. Because no need to find a hosting for images to keep them for your project

There might be some news coming in the next months on this side. In the meantime, we could add this button.

**Comment by ran4erep** [comment]
that would be nice, it's a really good idea about `Advanced options` button

**Comment by 4ian** [comment]
Two things:

- There is definitely a need for hosting resources. We'll have something to help this in the short term future.
- Base64 works, but it's a "hack", in the sense it was never officially supported. You can use it for now, but be prepared to change this in the future once we have something new :)
